src/components/Datepicker/Datepicker.js

```diff
@@ -146,46 +255,71 @@ export class Datepicker extends React.Component {
     );
   }
 
+  /**
+   * Renders a day
+   * @return {ReactElement} td
+   */
   renderDay(day, inRange, dayIndex) {
+    const { preselectedDate } = this.state;
     const classes = {
       [this.prefix('is-today')]: day.isSame(moment(), 'day'),
       [this.prefix('disabled-text')]: !inRange,
-      [this.prefix('is-selected')]: day.isSame(this.state.selectedDate, 'day'),
+      [this.prefix('is-selected')]: day.isSame(preselectedDate, 'day'),
     };
+
     return (
       <td key={dayIndex} className={classnames(classes)} headers={day.day()}>
-        <span className={this.prefix('day')} onClick={() => this.selectDay(day)}>
+        <span className={this.prefix('day')} onClick={() => this.onChangeDay(day)}>
           {day.date()}
         </span>
       </td>
     );
   }
 
   render() {
+    const { onToggle, isOpen, translations: { inputFieldLabel, inputFieldError } } = this.props;
```

Comment:
Avoid nesting destructuring. That becomes easily hard to read. Although it's more code, extract that to an extra line:

```js
const { onToggle, isOpen, translations } = this.props;
const { inputFieldLabel, inputFieldError } = translations;
```